sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_transport.py

```diff
@@ -653,5 +652,77 @@ def Transport(host, connect_timeout=None, ssl=False, **kwargs):
     Given a few parameters from the Connection constructor,
     select and create a subclass of _AbstractTransport.
     """
-    transport = SSLTransport if ssl else TCPTransport
+    transport_type = kwargs.pop('transport_type')
+    if transport_type == TransportType.AmqpOverWebsocket:
+        transport = WebSocketTransport
+    else:
+        transport = SSLTransport if ssl else TCPTransport
     return transport(host, connect_timeout=connect_timeout, ssl=ssl, **kwargs)
+
+class WebSocketTransport(_AbstractTransport):
+    def __init__(self, host, port=WEBSOCKET_PORT, connect_timeout=None, ssl=None, **kwargs
+        ):
+        self.sslopts = ssl if isinstance(ssl, dict) else {}
+        self._connect_timeout = connect_timeout
+        self._host = host
+        super().__init__(
+            host, port, connect_timeout, **kwargs
+            )
+        self.ws = None
+        self._http_proxy = kwargs.get('http_proxy', None)
+
+    def connect(self):
+        http_proxy_host, http_proxy_port, http_proxy_auth = None, None, None
+        if self._http_proxy:
+            http_proxy_host = self._http_proxy['proxy_hostname']
+            http_proxy_port = self._http_proxy['proxy_port']
+            username = self._http_proxy.get('username', None)
+            password = self._http_proxy.get('password', None)
+            if username or password:
+                http_proxy_auth = (username, password)
+        try:
+            from websocket import create_connection
+            self.ws = create_connection(
+                url="wss://{}/$servicebus/websocket/".format(self._host),
```

Comment:
This should be removed out of pyamqp and into the SDK layer.

Reply:
Can you clarify? Would you expect the users to provide a wss link by themselves?

Reply:
This doesn't look like it's generic - it looks specific to Azure services. So I would expect this value to be supplied by the EH or SB SDKs.
(So yes, customers of the pyamqp library would probably need to supply this - how was it being passed into uamqp?)

Reply:
hmm, That makes sense, but uamqp has been doing this instead of taking it from the client.

https://github.com/Azure/azure-uamqp-python/blob/master/src/wsio.pyx#L14

Reply:
An opportunity to improve ;)

Reply:
done